### PR TITLE
[RecoveryServices.Backup] Added Zone Redundancy for Recovery Service Vaults

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Added Zone Redundancy for Recovery Service Vaults

## Checklist

- [Yes] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [Yes] The title of the PR is clear and informative
- [Yes] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [Yes] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [N/A] If applicable, the changes made in the PR have proper test coverage
- [Yes] For public API changes to cmdlets:
    - [Yes] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [N/A] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by hiaga** [comment]
@isra-fel can you pls review.

File: src/RecoveryServices/RecoveryServices.Backup.Test/RecoveryServices.Backup.Test.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <PackageReference Include="Microsoft.Azure.Management.Compute" Version="39.0.0" />
     <PackageReference Include="Microsoft.Azure.Management.Network" Version="20.3.0" />
     <PackageReference Include="Microsoft.Azure.Management.RecoveryServices" Version="4.3.1-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.RecoveryServices.Backup" Version="4.1.3-preview" />
+	<PackageReference Include="Microsoft.Azure.Management.RecoveryServices.Backup" Version="4.1.3-preview" />
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Backup.Test/RecoveryServices.Backup.Test.csproj:17]: Please revert the unnecessary change.

File: src/RecoveryServices/RecoveryServices.Test/RecoveryServices.Test.csproj [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.RecoveryServices" Version="4.3.1-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.RecoveryServices.Backup" Version="4.1.3-preview" />
+	<PackageReference Include="Microsoft.Azure.Management.RecoveryServices.Backup" Version="4.1.3-preview" />
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices.Test/RecoveryServices.Test.csproj:15]: Please revert the unnecessary change.

File: src/RecoveryServices/RecoveryServices/Models/PSObjects.cs [hunk 3]
```diff
@@ -205,6 +205,7 @@ public VaultSettingsFilePath()
     public enum AzureRmRecoveryServicesBackupStorageRedundancyType
     {
         GeoRedundant = 1,
+        ZoneRedundant,
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/Models/PSObjects.cs:208]: Please also update the related cmdlet help file.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by hiaga** [comment]
@msJinLei : I've made the changes pls review again

**Comment by hiaga** [comment]
@erich-wang: can you pls review

**Review by msJinLei — APPROVED** [review]

**Pull request merged**